src/dbnode/storage/bootstrap/bootstrapper/base.go

```diff
@@ -268,12 +268,11 @@ func (b baseBootstrapper) logSuccessAndDetermineCurrResultsUnfulfilledAndNextBoo
 		// Set the modified result.
 		currResults.Results.Set(id, currResult)
 
-		// Set the next bootstrapper namespace run options if we need to bootstrap
-		// further time ranges.
-		if !nextNamespace.DataRunOptions.ShardTimeRanges.IsEmpty() ||
-			!nextNamespace.IndexRunOptions.ShardTimeRanges.IsEmpty() {
-			next.Namespaces.Set(id, nextNamespace)
-		}
+		// Always set the next bootstrapper namespace run options regardless of
+		// whether there are unfulfilled index/data shard time ranges.
+		// NB(bodu): We perform short circuiting directly in the peers bootstrapper and the
+		// commitlog bootstrapper should always run for all time ranges.
+		next.Namespaces.Set(id, nextNamespace)
```

Comment:
@notbdu curious about this change. it seems to have introduced a regression where we read the commit log even if there are no time ranges to fulfill. example of taking 11 mins to read a commit log without any range to fulfill.

```
{"level":"info","ts":1599641830.4700596,"msg":"bootstrap from source requested","bootstrapper":"commitlog","namespace":"default","numShards":85,"dataRange":0,"indexRange":0}
{"level":"info","ts":1599641830.4701018,"msg":"bootstrap from source ready after availability query","bootstrapper":"commitlog","namespace":"default","numShards":85,"dataRange":0,"indexRange":0}
{"level":"info","ts":1599642651.0040348,"msg":"bootstrapping from source completed successfully","bootstrapper":"commitlog","took":817.729679365,"namespace":"default","numShards":85,"dataRangeRequested":0,"dataRangeFulfilled":0,"indexRangeRequested":0,"indexRangeFulfilled":0,"numIndexBlocks":0}
```

cc: @robskillington

Reply:
We made this change so that the commit log bootstrapper would not be dependent on prior bootstrappers. The commit log bootstrapper will bootstrap the target shard time ranges irrespective of what has been previously fulfilled. 
```
// NB(r): Combine all shard time ranges across data and index
// so we can do in one go.
shardTimeRanges := result.NewShardTimeRanges()
// NB(bodu): Use TargetShardTimeRanges which covers the entire original target shard range
// since the commitlog bootstrapper should run for the entire bootstrappable range per shard.
shardTimeRanges.AddRanges(ns.DataRunOptions.TargetShardTimeRanges)
if ns.Metadata.Options().IndexOptions().Enabled() {
        shardTimeRanges.AddRanges(ns.IndexRunOptions.TargetShardTimeRanges)
}
```